File: classes/parte-7-paradigmas/115-funcional-ii-composicion-currying-y-aplicacion-parcial/implementaciones/go/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func doblar(x int) int      { return x * 2 }
func incrementar(x int) int { return x + 1 }

func main() {
	line, _ := bufio.NewReader(os.Stdin).ReadString('\n')
	n, _ := strconv.Atoi(strings.TrimSpace(line))
	fmt.Printf("resultado=%d\n", incrementar(doblar(n)))
}

File: classes/parte-7-paradigmas/115-funcional-ii-composicion-currying-y-aplicacion-parcial/implementaciones/c/main.c
#include <stdio.h>

long doblar(long x) { return x * 2; }
long incrementar(long x) { return x + 1; }

int main(void) {
    long n;
    if (scanf("%ld", &n) != 1) return 1;
    printf("resultado=%ld\n", incrementar(doblar(n)));
    return 0;
}
